src/device-state.coffee

```diff
@@ -675,24 +675,25 @@ module.exports = class DeviceState extends EventEmitter
 		@applyUnblocker?()
 		return
 
-	triggerApplyTarget: ({ force = false, delay = 0, initial = false } = {}) =>
-		if @applyInProgress
-			if !@scheduledApply?
+	triggerApplyTarget: ({ force = false, delay = 0, initial = false, isFromApi = false } = {}) =>
+		if @applyInProgress?
+			if !@scheduledApply? || isFromApi
 				@scheduledApply = { force, delay }
+				if isFromApi
+					@applyInProgress.cancel()
```

Comment:
I think we need to call `Promise.config({ cancellation: true })` somewhere for this to actually work. I'd recommend putting it in the app's entrypoint, app.coffee.

Reply:
Yes I just read through the bluebird promise documentation for cancellation. Sorry for the oversight.